12-offer-encoding.md

```diff
@@ -0,0 +1,823 @@
+   in other forms (e.g. inside the lightning network itself).
+2. The signature applying to the entire invoice makes it impossible
+   to prove an invoice without revealing its entirety.
+3. Fields cannot generally be extracted for external use: the `h`
+   field was a boutique extraction of the `d` field, only.
+4. The lack of 'it's OK to be odd' rule makes backwards compatibility
+   harder.
+5. The 'human-readable' idea of separating amounts proved fraught:
+   `p` was often mishandled, and amounts in pico-bitcoin are harder
+   than the modern satoshi-based counting.
+6. The bech32 encoding was found to have an issue with extensions,
+   which means we want to replace or discard it anyway.
+7. The `payment_secret` designed to prevent probing by other nodes in
+   the path was only useful if the invoice remained private between the
+   payer and payee.
+8. Invoices must be given per-user, and are actively dangerous if two
+   payment attempts are made for the same user.
+
+
+# Payment Flow Scenarios
+
+Here we use "user" as shorthand for the individual user's lightning
+node, and "merchant" as the shorthand for the node of someone who is
+selling or has sold something.
+
+There are two basic payment flows supported by BOLT 12:
+
+The general user-pays-merchant flow is:
+1. A merchant publishes an *offer* ("send me money"), such as on a web page or a QR code.
+2. Every user requests a unique *invoice* over the lightning network
+   using an *invoice_request* message.
+3. The merchant replies with the *invoice*.
+4. The user makes a payment to the merchant indicated by the invoice.
+
+The merchant-pays-user flow (e.g. ATM or refund):
+1. The merchant provides a user-specific *offer* ("take my money") in a web page or QR code,
+   with an amount (for a refund, also a reference to the to-be-refunded
+   invoice).
+2. The user sends an *invoice* for the amount in the *offer* (for a
+   refund, a proof that they requested the original)
+3. The merchant makes a payment to the user indicated by the invoice.
+
+## Payment Proofs and Payer Proofs
+
+Note that the normal lightning "proof of payment" can only demonstrate that an
+invoice was paid (by showing the preimage of the `payment_hash`), not who paid
+it.  The merchant can claim an invoice was paid, and once revealed, anyone can
+claim they paid the invoice, too.[1]
+
+Providing a key in *invoice_request* allows a user to prove that they were the one
+to request the invoice.  In addition, the Merkle construction of the BOLT 12
+invoice signature allows the user to selectively reveal fields of the invoice
+in case of dispute.
+
+# Encoding
+
+Each of the forms documented here are in
+[TLV](01-messaging.md#type-length-value-format) format.
+
+The supported ASCII encoding is the human-readable prefix, followed by a
+`1`, followed by a bech32-style data string of the TLVs in order,
+optionally interspersed with `+` (for indicating additional data is to
+come).
+
```

Comment:
why not use bech32m instead here that fixes the bug in the checksum?

Reply:
Because we don't need the checksum. That's all written in the "Rationale" section a few lines below.

Reply:
to be fair, the rationale says it's part of the reason it's not including the checksum. If it's irrelevant to choosing a bech32(m) checksum or not, that sentence should be dropped